### PR TITLE
Particle emitter - part1

### PR DESCRIPTION
**Pull request by caguero**
This is pretty much all the boilerplate for the upcoming particle emitter class.

### CODE REVIEW

**Comment by iche033** [comment]
just noticed that there are linker errors on windows build. In the past we had to add empty classes like [this](https://github.com/ignitionrobotics/ign-rendering/blob/master/src/GaussianNoisePass.cc) to fix these errors. I this case, we may need to add a ParticleEmitter.cc

File: include/ignition/rendering/ParticleEmitter.hh [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  namespace rendering
+  {
+    inline namespace IGNITION_RENDERING_VERSION_NAMESPACE {
+    //
+    /// \class ParticleEmitter ParticleEmitter.hh
+    /// ignition/rendering/ParticleEmitter.hh
+    //
+    /// \brief Class to manage a particle emitter.
+    class IGNITION_RENDERING_VISIBLE ParticleEmitter :
+      public virtual CompositeVisual
```
[the remark this answers is not in the document]
Comment (caguero) [reply include/ignition/rendering/ParticleEmitter.hh]: > the base class can be `Visual` (instead of `CompositeVisual`)

Fixed in 6f0a2e7.

**Comment by caguero** [comment]
> just noticed that there are linker errors on windows build. In the past we had to add empty classes like [this](https://github.com/ignitionrobotics/ign-rendering/blob/master/src/GaussianNoisePass.cc) to fix these errors. In this case, we may need to add a ParticleEmitter.cc

Let's wait and see if 93b9116 fixes the linking errors.

**Comment by caguero** [comment]
> > just noticed that there are linker errors on windows build. In the past we had to add empty classes like [this](https://github.com/ignitionrobotics/ign-rendering/blob/master/src/GaussianNoisePass.cc) to fix these errors. In this case, we may need to add a ParticleEmitter.cc
> 
> Let's wait and see if [93b9116](https://github.com/ignitionrobotics/ign-rendering/commit/93b9116e883b4bfedc84c1edbf17196a927963c8) fixes the linking errors.

Seems fixed:
https://build.osrfoundation.org/job/ign_rendering-pr-win/826/

**Pull request merged**